### PR TITLE
X3D deserializer: accept commas as MF value delimiters

### PR DESCRIPTION
**Pull request by andreasplesch**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

see https://github.com/jscad/OpenJSCAD.org/discussions/1241#discussioncomment-6890943

I used a few scripts to refactor the parsing of MF field attribute values and to help with catching all occurrences. The tests run ok and the generated jscad script is identical to inputs without commas. It is still possible that some case was missed.

### CODE REVIEW

File: packages/io/x3d-deserializer/src/objects.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+//sed 's|= \(element\..*[^\.]*\).trim().split.*map.*|= parseNumbers(\1)|' objects.js > objects_2.js
+//sed 's|= \(element\..*[^\.]*\).trim().split.*-1.*|= parseIndices(\1)|' objects_2.js | sed 's|.map((index) => index.trim().split(/ +/).map((v) => parseFloat(v)))||' > objects_3.js
```
Comment (andreasplesch) [inline packages/io/x3d-deserializer/src/objects.js:2]: will be removed

**Comment by andreasplesch** [comment]
Node 14 apparently does not have the replaceAll method for strings. Let me see what the node 14 testing is about. Probably there is a simple replacement.

**Comment by andreasplesch** [comment]
see #1137 . Apparently Node 14 is still widely in use. Let me change to `.replace( /,/g , " " )`.

File: packages/io/x3d-deserializer/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,9 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+- allow commas as delimiter in MF fields
```
Comment (z3dev) [inline packages/io/x3d-deserializer/CHANGELOG.md:6]: Actually, the change logs are automatically updated when new releases are published to NPM. So, this change is not necessary.

there's some information here...

https://github.com/jscad/OpenJSCAD.org/wiki/Admin-:--Merging-Pull-Requests

**Review by z3dev — APPROVED** [review]
@andreasplesch very nice set of changes. The code looks really clean after adding this two new functions.

File: packages/io/x3d-deserializer/CHANGELOG.md (hunk 2 above)
Comment (andreasplesch) [reply packages/io/x3d-deserializer/CHANGELOG.md:6]: Ok. Thanks for the info.

**Pull request merged**